test/lib/test_part.ml

```diff
@@ -1,45 +1,48 @@
-      | Compat_attr (n, sep_indent) ->
-          Fmt.pf fmt "Compat_attr (%S, %S)" n sep_indent
-      | Part_begin (n, sep_indent) ->
-          Fmt.pf fmt "Part_begin (%S, %S)" n sep_indent
-      | Part_end -> Fmt.pf fmt "Part_end"
-      | File_end -> Fmt.pf fmt "File_end"
-    in
-    Alcotest.testable pp equal
-end
+let parse_parts =
+  let equal = ( = ) in
+  let pp fmt = function
+    | Parse_parts.Content s -> Fmt.pf fmt "Content: %S" s
+    | Compat_attr { name; sep_indent } ->
+        Fmt.pf fmt "Compat_attr {name=%S; sep_indent=%S}" name sep_indent
+    | Part_begin { name; sep_indent } ->
+        Fmt.pf fmt "Part_begin {name=%S; sep_indent=%S}" name sep_indent
+    | Part_end -> Fmt.pf fmt "Part_end"
+  in
+  Alcotest.testable pp equal
 
 let test_of_line =
-  let make_test ~line ~expected () =
-    let test_name =
-      Printf.sprintf "parse_line: %S"
-        (match line with Ok s -> s | Error `End_of_file -> "EOF")
-    in
+  let make_test ~line ~expected =
+    let test_name = Printf.sprintf "parse: %S" line in
     let test_fun () =
-      let actual = Parse_parts.parse_line line in
-      Alcotest.(check Testable.parse_parts_part_decl) test_name expected actual
+      let actual = Parse_parts.parse line in
+      Alcotest.(check (Testable.errormsg (list parse_parts)))
+        test_name expected actual
     in
     (test_name, `Quick, test_fun)
   in
   [
-    make_test ~line:(Error `End_of_file) ~expected:File_end ();
-    make_test ~line:(Ok "foo") ~expected:(Normal "foo") ();
-    make_test ~line:(Ok "    (* $MDX part-begin=bar *)    ")
-      ~expected:(Part_begin ("bar", "    "))
-      ();
-    make_test ~line:(Ok "(* $MDX part-begin=bar     ")
-      ~expected:(Normal "(* $MDX part-begin=bar     ") ();
-    make_test ~line:(Ok "   (* $MDX part-end   *)   ") ~expected:Part_end ();
-    make_test ~line:(Ok "    [@@@part \"foobar\"]    ")
-      ~expected:(Compat_attr ("foobar", "    "))
-      ();
+    make_test ~line:"" ~expected:(Ok [ Content "" ]);
+    make_test ~line:"foo" ~expected:(Ok [ Content "foo" ]);
+    make_test ~line:"    (* $MDX part-begin=bar *)    "
+      ~expected:(Ok [ Part_begin { name = "bar"; sep_indent = "    " } ]);
+    make_test ~line:"(* $MDX part-begin=bar     "
+      ~expected:(Ok [ Content "(* $MDX part-begin=bar     " ]);
+    make_test ~line:"   (* $MDX part-end   *)   " ~expected:(Ok [ Part_end ]);
+    make_test ~line:"    [@@@part \"foobar\"]    "
+      ~expected:(Ok [ Compat_attr { name = "foobar"; sep_indent = "    " } ]);
+    make_test ~line:"[@@@foo \"bar\"]"
+      ~expected:(Ok [ Content "[@@@foo \"bar\"]" ]);
+    make_test ~line:"(* $MDX foo *)"
+      ~expected:
+        (Error (`Msg "'(* $MDX foo *)' is not a valid ocaml delimiter for mdx."));
+    make_test ~line:"(* $MDX part-end=foo *)"
+      ~expected:
+        (Error
+           (`Msg
+             "'part-end' delimiter does not accept a value. Please write '(* \
+              $MDX part-end *)' instead."));
```

Comment:
Not important but since this part of the API is now internal, wouldn't it better to test it with ppx_expect instead ? So we can not export this module anymore.